db/batch/batch_impl.go

```diff
@@ -359,6 +364,10 @@ func (cb *cachedBatch) ResetSnapshots() {
 	cb.lock.Lock()
 	defer cb.lock.Unlock()
 
+	// ignore reset snapshots if the snapshot number is less than threshold
+	if cb.tag <= resetSnapshotIgnoreThreshold {
+		return
```

Comment:
Could you elaborate it?
My understanding: snapshot id is just an id. Inside of evm, it will be called as follows:
```
  id := snapshot()
  some other code
  id2 := snapshot()
  some other code
  if need to revert id
      revertSnapshot(id)
  
```
Thus, the real value of the `id` doesn't matter.

Reply:
I have checked that the SerializeQueue() function wouldn't be impacted by the improvement, so no hard fork is needed. However, it would be more straightforward if there were tests to ensure that the improvement does not impact the function.